lib/cancan/ability.rb

```diff
@@ -302,7 +302,11 @@ def extract_subjects(subject)
 
     def alternative_subjects(subject)
       subject = subject.class unless subject.is_a?(Module)
-      [:all, *subject.ancestors, subject.class.to_s]
+      if subject.methods.include? :subclasses
```

Comment:
when does it happen that it does not have the subclasses method? can you use `if subject.respond_to?(:subclasses)`?

Reply:
Some of the subjects are Ranges which do not have subclasses